frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/deprecated-operator-warnings.cy.ts

```diff
@@ -0,0 +1,227 @@
+    );
+    cy.exec(
+      `oc delete ${testDeprecatedCatalogSource.kind} ${testDeprecatedCatalogSource.metadata.name} -n ${testDeprecatedCatalogSource.metadata.namespace}`,
+    );
+    checkErrors();
+  });
+
+  it('verify deprecated operator warning badge on the OperatorHub tile', () => {
+    cy.visit(
+      `/k8s/ns/${testDeprecatedCatalogSource.metadata.namespace}/operators.coreos.com~v1alpha1~CatalogSource/test-community-operator-deprecation`,
+    );
+    cy.log('verify the test-community-operator-deprecation CatalogSource is in "READY" status');
+    cy.byTestSelector('details-item-value__Status', TIMEOUT).should('have.text', 'READY');
+
+    cy.log('visit OperatorHub');
+    cy.visit('/operatorhub/all-namespaces');
+
+    cy.log('filter by the group name');
+    cy.byTestID('source-community-operators-for-testing-deprecation').click();
+
+    cy.log('filter by the operator name');
+    cy.byTestID('search-operatorhub').type(testOperatorName);
+    cy.get('.co-catalog-tile', TIMEOUT).its('length').should('eq', 1);
+
+    cy.log('verify the Deprecated badge on Kiali Community Operator tile');
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_BADGE_ID)
+      .contains(deprecatedBadge)
+      .should('be.visible');
+  });
+
+  it('verify deprecated operator warnings on the Install Operator modal', () => {
+    cy.log('visit Install Operator modal');
+    cy.visit(
+      '/operatorhub/all-namespaces?keyword=kia&details-item=kiali-test-community-operator-deprecation-openshift-marketplace&channel=stable&version=1.83.0',
+    );
+    cy.log('verify the Deprecated badge on Kiali Community Operator logo');
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_BADGE_ID)
+      .contains(deprecatedBadge)
+      .should('be.visible');
+
+    cy.log('verify the package deprecation warning exists');
+    operator.modalElementShouldBeVisible(
+      DEPRECATED_OPERATOR_WARNING_PACKAGE_ID,
+      deprecatedPackageMessage,
+    );
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_CHANNEL_ID)
+      .contains(deprecatedChannelMessage)
+      .should('not.exist');
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_VERSION_ID)
+      .contains(deprecatedVersionMessage)
+      .should('not.exist');
+
+    cy.log('verify the channel deprecation warnings exists');
+    cy.get('.co-operator-channel__select').children('.pf-v5-c-select__toggle').click();
+    cy.byTestID('channel-option-alpha')
+      .children()
+      .children('[data-test="deprecated-operator-warning-channel-icon"]')
+      .should('exist');
+    cy.byTestID('channel-option-alpha').click();
+
+    operator.modalElementShouldBeVisible(
+      DEPRECATED_OPERATOR_WARNING_CHANNEL_ID,
+      deprecatedChannelMessage,
+    );
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_VERSION_ID)
+      .contains(deprecatedVersionMessage)
+      .should('not.exist');
+
+    cy.log('verify the version deprecation warning exists');
+    cy.get('.co-operator-version__select').children('.pf-v5-c-select__toggle').click();
```

Comment:
This `cy.byTestID('channel-select').children('.pf-v5-c-select__toggle').click();` seems not to work for me. I added `channel-selec`t to the `SelectDeprecated` component expecting it to be available where the `co-operator-channel__select` is, but instead, it is in the `ul` element, which doesn't trigger the select dropdown. Doesn't this work for you?

Reply:
Yeah, I'm fine with using the class selector if PF doesn't provide any way to pass through a data attribute to the toggle button. We should probably update this when we migrate away from the deprecated component.